gapic/src/main/com/google/gapid/widgets/TabComposite.java

```diff
@@ -0,0 +1,1173 @@
+      boolean firstDone = x1 < 0 || y1 < 0, secondDone = x2 < 0 || y2 < 0;
+      if (firstDone && secondDone) {
+        return;
+      }
+
+      for (Element child : children) {
+        boolean first = child.contains(x1, y1);
+        boolean second = child.contains(x2, y2);
+        if (first && second) {
+          child.redrawBar(x1, y1, x2, y2);
+          return;
+        } else if (first) {
+          child.redrawBar(x1, y1, -1, -1);
+          if (secondDone) {
+            return;
+          }
+          firstDone = true;
+        } else if (second) {
+          child.redrawBar(x2, y2, -1, -1);
+          if (firstDone) {
+            return;
+          }
+          secondDone = true;
+        }
+      }
+    }
+
+    @Override
+    protected Hover find(Group parent, int index, int mx, int my, boolean includeTrailing) {
+      for (int i = 0; i < children.size(); i++) {
+        Element child = children.get(i);
+        if (mx >= child.x && mx < child.x + child.w &&
+            my >= child.y && my < child.y + child.h) {
+          return child.find(this, i, mx, my, includeTrailing);
+        }
+      }
+      return Hover.NONE;
+    }
+
+    protected abstract void moveSeparator(int index, int sx, int sy);
+
+    @Override
+    protected MergeState merge() {
+      for (ListIterator<Element> it = children.listIterator(); it.hasNext(); ) {
+        Element current = it.next();
+        MergeState state = current.merge();
+        if (state == MergeState.REMOVE) {
+          it.remove();
+        } else if (state == MergeState.DO_NOTHING) {
+          // Do nothing.
+        } else {
+          state.replacement.weight = current.weight;
+          it.set(state.replacement);
+        }
+      }
+
+      switch (children.size()) {
+        case 0: return MergeState.REMOVE;
+        case 1: return MergeState.replace(children.get(0));
+        default: return MergeState.DO_NOTHING;
+      }
+    }
+  }
+
+  private static enum Location {
+    Left, Right, Top, Bottom;
+
+    public Rectangle highlight(Folder f) {
+      switch (this) {
+        case Left:   return new Rectangle(f.x, f.y, f.w / 3, f.h);
+        case Right:  return new Rectangle(f.x + 2 * f.w / 3, f.y, f.w / 3, f.h);
+        case Top:    return new Rectangle(f.x, f.y, f.w, f.h / 3);
+        case Bottom: return new Rectangle(f.x, f.y + 2 * f.h / 3, f.w, f.h / 3);
+        default: throw new AssertionError();
+      }
+    }
+  }
+
+  private class HorizontalGroup extends Group {
+    public HorizontalGroup(int weight) {
+      super(weight, false, true);
+    }
+
+    @Override
+    protected void setChildBounds(Set<Control> controls, int weightTotal) {
+      int cw = w - (children.size() - 1) * FOLDER_MARGIN;
+      if (weightTotal <= 0) {
+        int fw = cw / children.size();
+        int rem = cw % children.size();
+        for (int i = 0, fx = x; i < children.size(); i++) {
+          int add = rem-- > 0 ? 1 : 0;
```

Comment:
For the sake of clarity, could `rem--` be moved to an other line?:

```
int add = rem > 0 ? 1 : 0;
rem = rem - 1;
```

Reply:
moved `rem--` to the for line